core/src/main/java/bisq/core/offer/OpenOfferManager.java

```diff
@@ -871,39 +871,46 @@ private void maybeUpdatePersistedOffers() {
     ///////////////////////////////////////////////////////////////////////////////////////////
 
     private void republishOffers() {
-        int size = openOffers.size();
-        final ArrayList<OpenOffer> openOffersList = new ArrayList<>(openOffers.getList());
-        if (!stopped) {
-            stopPeriodicRefreshOffersTimer();
-            for (int i = 0; i < size; i++) {
-                // we delay to avoid reaching throttle limits
-
-                long delay = 700;
-                final long minDelay = (i + 1) * delay;
-                final long maxDelay = (i + 2) * delay;
-                final OpenOffer openOffer = openOffersList.get(i);
-                UserThread.runAfterRandomDelay(() -> {
-                    if (openOffers.contains(openOffer)) {
-                        String id = openOffer.getId();
-                        if (id != null && !openOffer.isDeactivated())
-                            republishOffer(openOffer);
-                    }
+        if (stopped) {
+            return;
+        }
 
-                }, minDelay, maxDelay, TimeUnit.MILLISECONDS);
-            }
-        } else {
-            log.debug("We have stopped already. We ignore that republishOffers call.");
+        List<OpenOffer> openOffersList = new ArrayList<>(openOffers.getList());
+        republishOffers(openOffersList);
+
+        stopPeriodicRefreshOffersTimer();
+    }
+
+    private void republishOffers(List<OpenOffer> list) {
+        if (list.isEmpty()) {
+            return;
         }
+
+        OpenOffer openOffer = list.remove(0);
+        if (!openOffers.contains(openOffer) || openOffer.isDeactivated()) {
```

Comment:
Shouldn't this be `!openOffer.isDeactivated()` since it's being republished. That was the check before this change.

Reply:
No it is correct, we call our recursive method with the list where we have removed that offer. But the following lines neede to be in a else branch.
I made a PR where I invert the if check and add a comment.

Reply:
Yes, understood after the fix. Also the fix handles the recursion issue so I think it's good to merge.